pkg/node_reputation/node_reputation_handler.go

```diff
@@ -0,0 +1,280 @@
+	res := proto.UpdateReply_UPDATE_FAILED
+
+	selectNodeStmt := genWhereStatement(selectAllStmt, nodeNameColumn, equal, in.NodeName)
+	row, err := endianReputation(db, selectNodeStmt)
+	if err != nil {
+		res = proto.UpdateReply_UPDATE_FAILED
+	}
+
+	if row.nodeName != in.NodeName {
+		row.source = in.Source
+		row.nodeName = in.NodeName
+	}
+	newRecord := row
+
+	switch toColumn(in.ColumnName) {
+	case uptimeColumn:
+		val, err := strconv.ParseInt(in.ColumnValue, 10, 64)
+		if err != nil {
+			res = proto.UpdateReply_UPDATE_FAILED
+		} else {
+			res = proto.UpdateReply_UPDATE_SUCCESS
+			newRecord = newRecord.morphism(uptimeColumn, overWrite, val)
+		}
+		morph := []nodeReputationRecord{newRecord}
+		insertRows(db, morph, insertStmt)
+
+	case auditSuccessColumn:
+		val, err := strconv.ParseInt(in.ColumnValue, 10, 64)
+		if err != nil {
+			res = proto.UpdateReply_UPDATE_FAILED
+		} else {
+			res = proto.UpdateReply_UPDATE_SUCCESS
+			newRecord = newRecord.morphism(auditSuccessColumn, overWrite, val)
+		}
+		morph := []nodeReputationRecord{newRecord}
+		insertRows(db, morph, insertStmt)
+
+	case auditFailColumn:
+		val, err := strconv.ParseInt(in.ColumnValue, 10, 64)
+		if err != nil {
+			res = proto.UpdateReply_UPDATE_FAILED
+		} else {
+			res = proto.UpdateReply_UPDATE_SUCCESS
+			newRecord = newRecord.morphism(auditFailColumn, overWrite, val)
+		}
+		morph := []nodeReputationRecord{newRecord}
+		insertRows(db, morph, insertStmt)
+
+	case latencyColumn:
+		val, err := strconv.ParseInt(in.ColumnValue, 10, 64)
+		if err != nil {
+			res = proto.UpdateReply_UPDATE_FAILED
+		} else {
+			res = proto.UpdateReply_UPDATE_SUCCESS
+			newRecord = newRecord.morphism(latencyColumn, overWrite, val)
+		}
+		morph := []nodeReputationRecord{newRecord}
+		insertRows(db, morph, insertStmt)
+
+	case amountOfDataStoredColumn:
+		val, err := strconv.ParseInt(in.ColumnValue, 10, 64)
+		if err != nil {
+			res = proto.UpdateReply_UPDATE_FAILED
+		} else {
+			res = proto.UpdateReply_UPDATE_SUCCESS
+			newRecord = newRecord.morphism(amountOfDataStoredColumn, overWrite, val)
+		}
+		morph := []nodeReputationRecord{newRecord}
+		insertRows(db, morph, insertStmt)
+
+	case falseClaimsColumn:
+		val, err := strconv.ParseInt(in.ColumnValue, 10, 64)
+		if err != nil {
+			res = proto.UpdateReply_UPDATE_FAILED
+		} else {
+			res = proto.UpdateReply_UPDATE_SUCCESS
+			newRecord = newRecord.morphism(falseClaimsColumn, overWrite, val)
+		}
+		morph := []nodeReputationRecord{newRecord}
+		insertRows(db, morph, insertStmt)
+
+	case shardsModifiedColumn:
+		val, err := strconv.ParseInt(in.ColumnValue, 10, 64)
+		if err != nil {
+			res = proto.UpdateReply_UPDATE_FAILED
+		} else {
+			res = proto.UpdateReply_UPDATE_SUCCESS
+			newRecord = newRecord.morphism(shardsModifiedColumn, overWrite, val)
+		}
+		morph := []nodeReputationRecord{newRecord}
+		insertRows(db, morph, insertStmt)
+
+	default:
```

Comment:
in this file and in other files and pull requests, there are an awful lot of switch statements. i wonder if there's a way to reduce all of the one-off per-column handling. it would be great to minimize the number of files that have to change if we add a new column